html/js/app/ui/event_alert_indicators.js

```diff
@@ -1,73 +1,86 @@
-                    found = found || arn == previous.resource_arn;
-                    if (found) {
-                        break;
-                    }
-                }
-                if (!found) {
-                    inactive_nodes.push(previous.resource_arn);
-                }
-            }
-
-            // 'unalert' the nodes that are no longer alerting
-            for (let arn of inactive_nodes) {
-                var node = model.nodes.get(arn);
-                if (node) {
-                    node.alerting = false;
-                    var selected = node.render.normal_selected();
-                    var unselected = node.render.normal_unselected();
-                    if (selected != node.image.selected || unselected != node.image.unselected) {
-                        node.image.selected = selected;
-                        node.image.unselected = unselected;
-                        model.nodes.update(node);
-                        var matches = diagrams.have_all([node.id]);
-                        // console.log(matches);
-                        for (let diagram of matches) {
-                            diagram.nodes.update(node);
-                            diagram.alert(false);
-                        }
-                    }
-                }
-            }
-        };
-
-        event_alerts.add_callback(updateEventAlertState);
-    });
\ No newline at end of file
+       define(["jquery", "lodash", "app/model", "app/events", "app/ui/diagrams"],
+       function($, _, model, event_alerts, diagrams) {
+   
+           const updateAlertHandler = node => {
+               const selected = node.degraded ? node.render.degraded_selected()
+                   : node.render.alert_selected();
+               const unselected = node.degraded ? node.render.degraded_unselected()
+                   : node.render.alert_unselected();
+   
+               if (selected != node.image.selected || unselected != node.image.unselected) {
+                   node.image.selected = selected;
+                   node.image.unselected = unselected;
+                   model.nodes.update(node);
+   
+                   const matches = diagrams.have_all([node.id]);
+   
+                   for (let diagram of matches) {
+                       diagram.nodes.update(node);
+                       diagram.alert(true);
+                   }
+               }
+           };
+   
+           const updateEventAlertState = (current_alerts, previous_alerts) => {
+               // iterate through current 'set' alerts
+               const alerting_nodes = [];
+               const inactive_nodes = [];
+               const degraded_nodes = [];
+   
+               for (let item of current_alerts) {
+   
+                   const node = model.nodes.get(item.resource_arn);
+   
+                   if (node) {
+                       node.alerting = true;
+                    //    node.degraded =  true;
```

Comment:
@JimTharioAmazon @morjoan I know you guys have more ways of testing this, but I left this here in case you want to quickly test/update a degraded node. 
By uncommenting this line and commenting the next line, you will see nodes in a degraded state.